File: Recursion/643-maximum-average-subarray-i/maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int left=0,right=0;
        int sum =0;
        double ans=-1e18;
        int n = nums.size();
        if(n==1) return nums[0]/1.0;
        while(right<nums.size()){
            sum+=nums[right];
            right++;
            if(right-left==k){
                double avg = double(sum)/k;
                ans = max(avg,ans);
                sum-=nums[left];
                left++;
            }
        }
        return ans;
        
    }
};

File: Linked_List/643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {

        int left=0,right=0;



        double maxi=Integer.MIN_VALUE;

        int n = nums.length;

        if(n==1) return nums[0]/1.0;
        double sum=0;

        while(right<n){

            sum+=nums[right];

            if(right-left+1==k){

               maxi = Math.max(maxi,sum/k);
                sum-=nums[left];
                left++;
            }

            right++;
        }
        return maxi;
        
    }
}